test_package/conanfile.py

```diff
@@ -16,5 +15,17 @@ def build(self):
         cmake.build()
 
     def test(self):
-        bin_path = os.path.join("bin", "test_package")
-        self.run(bin_path, run_environment=True)
+        ## check if Binary can be executed
+        p = subprocess.Popen(["file", os.path.join("bin", "test_package")], stdout=subprocess.PIPE, shell=True)
+        output = p.communicate()[0]
+        
+        arch = platform.architecture()
+        
+        ## only Execute if it's possible
+        ## it's not releay a nice way..
+        if arch == "64bit":
+            if "64-bit" in output and "x86-64" in output: 
+                self.run("LD_LIBRARY_PATH=%s %s"%(os.environ.get('LD_LIBRARY_PATH', ''), os.path.join("bin","test_package")), run_environment=True)
+        elif arch == "32bit":
+            if "32-bit" in output and "x86" in output:
+                self.run("LD_LIBRARY_PATH=%s %s"%(os.environ.get('LD_LIBRARY_PATH', ''), os.path.join("bin","test_package")), run_environment=True)
```

Comment:
with `run_environment=True` setting `LD_LIBRARY_PATH` is no longer necessary, remove it to simplify code